qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -206,9 +207,24 @@ def _marginalized_component_data(self, composite_data: List[Dict]) -> List[List[
                         sub_data["counts"] = datum["counts"]
                 if "memory" in datum:
                     if composite_clbits is not None:
-                        sub_data["memory"] = (
-                            np.array(datum["memory"])[composite_clbits[i]]
-                        ).tolist()
+                        # level 2
+                        if isinstance(datum["memory"][0], str):
+                            num_cbits = 1 + max(
+                                cbit for cbit_list in composite_clbits for cbit in cbit_list
+                            )
+                            header = {"memory_slots": num_cbits}
+                            marginalized_memory = []
+                            for shot in datum["memory"]:
+                                shot = format_counts_memory(shot, header)[::-1]
+                                marginalized_memory.append(
+                                    "".join(shot[idx] for idx in composite_clbits[i])[::-1]
+                                )
+                            sub_data["memory"] = marginalized_memory
+                        # level 1
```

Comment:
Is it possible to add a test for level-1 marginalization?

Reply:
See here: [5b3eee0](https://github.com/Qiskit/qiskit-experiments/pull/801/commits/5b3eee0fb84770a812ad042ee974dc4f2d33ca22) This also required adding a fix to be able to handle single-shot memory vs averaged memory.